algebird-core/src/main/scala/com/twitter/algebird/Preparer.scala

```diff
@@ -0,0 +1,132 @@
+package com.twitter.algebird
+
+trait Preparer[A, T, +This[A, T] <: Preparer[A, T, This]] extends HasMonoidAggregate[A, T] {
```

Comment:
need tests for this to exercise and demo the code.

Also, the scala docs should exists.